moto/iam/responses.py

```diff
@@ -87,6 +87,12 @@ def list_instance_profiles(self):
         template = self.response_template(LIST_INSTANCE_PROFILES_TEMPLATE)
         return template.render(instance_profiles=profiles)
 
+    def list_instance_profiles_for_role(self, role_name=None):
+        profiles = iam_backend.get_instance_profiles_for_role(role_name=role_name)
```

Comment:
Instead of special-casing `list_instance_profiles_for_role`, this method already has access to the querystring so we can  just do something like
```python
role_name = self.querystring['RoleName'][0])
```

or using a little helper we've added:

```python
role_name = self._get_param('RoleName')
```

Apologies that none of the internals are very well documented for this type of thing.

If we can get that cleaned up, I think everything else looks good.